### PR TITLE
Added logic to render 404 if user accesses order that does not belong…

### PR DESCRIPTION
**Pull request by limsammy**
Related Issue: #120 #111 #114 #110 #118 #90 

Summary: Renders 404 if user accesses order not belonging to them

Author(s): sam

### CODE REVIEW

File: app/views/orders/index.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 <% current_user.orders.each do |order| %>
 	<li>
 		Order id: <%= order.id %><br>
-		<%= order.created_at %><br>
+		<%= order.created_at.to_formatted_s(:long_ordinal) %><br>
```
Comment (randy-darsh) [inline app/views/orders/index.html.erb:7]: dig the time format

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -6,8 +6,12 @@ def index
 	end
 
 	def show
-		@order = Order.find(params[:id])
-		@items = @order.items.distinct
+		if Order.find(params[:id]).user_id == current_user.id
+			@order = Order.find(params[:id])
+			@items = @order.items.distinct
+		else
```
Comment (amhursh) [inline app/controllers/orders_controller.rb:12]: Cool. Should protect non-user orders from access.

File: app/controllers/carts_controller.rb [hunk 3]
```diff
@@ -17,9 +17,18 @@ def create
...
+      redirect_to cart_path
+    elsif params[:Quantity] == '0'
+      item = Item.find(params[:item_id])
+      session[:cart].delete(item.id.to_s)
+      redirect_to cart_path
+    else
+      item = Item.find(params[:item_id])
+      @cart.update_item(item.id, params[:Quantity])
+      redirect_to cart_path
+    end
```
Comment (randy-darsh) [inline app/controllers/carts_controller.rb:31]: this was a necessary piece. good job

File: app/controllers/carts_controller.rb [hunk 4]
```diff
@@ -17,9 +17,18 @@ def create
...
+      flash[:failure] = "Cannot enter negative number!"
+      redirect_to cart_path
+    elsif params[:Quantity] == '0'
+      item = Item.find(params[:item_id])
+      session[:cart].delete(item.id.to_s)
+      redirect_to cart_path
+    else
+      item = Item.find(params[:item_id])
+      @cart.update_item(item.id, params[:Quantity])
+      redirect_to cart_path
```
Comment (amhursh) [inline app/controllers/carts_controller.rb:30]: Awesome. We can think about a refactor when we are doing controller refactors

**Review by amhursh — APPROVED** [review]

**Comment by randy-darsh** [comment]
looks good. merging to dev

**Pull request merged**